riscv/mmu.h

```diff
@@ -187,6 +211,21 @@ class mmu_t
     })
   }
 
+  // for shadow stack amoswap
+  template<typename T>
+  T ssamoswap(reg_t addr, reg_t value) {
+    convert_load_traps_to_store_traps({
+      bool forced_virt = false;
+      bool hlvx = false;
+      bool lr = false;
+      bool ss_access = true;
+      store_slow_path(addr, sizeof(T), nullptr, {forced_virt, hlvx, lr, ss_access}, false, true);
+      auto return_address = ss_load<T>(addr);
```

Comment:
For `ssamoswap`, lets just call this `data` instead of `return_address`. The typical use of `ssamoswap` is to operate on checkpoints stored on the shadow stack.